lib/addon-manager-proxy.js

```diff
@@ -27,33 +27,6 @@ class AddonManagerProxy extends EventEmitter {
     this.pluginClient = pluginClient;
     this.userProfile = pluginClient.userProfile;
     this.onUnload = null;
-
-    this.on(Constants.PROPERTY_CHANGED, (property) => {
-      DEBUG && console.log('AddonManagerProxy: Got',
-                           Constants.PROPERTY_CHANGED, 'notification for',
-                           property.name, 'value:', property.value);
-      this.sendPropertyChangedNotification(property);
-    });
-
-    this.on(Constants.ACTION_STATUS, (action) => {
-      DEBUG && console.log('AddonManagerProxy: Got',
-                           Constants.ACTION_STATUS, 'notification for',
-                           action.name, 'status:', action.status);
-      this.sendActionStatusNotification(action);
-    });
-
-    this.on(Constants.EVENT, (event) => {
-      DEBUG && console.log('AddonManagerProxy: Got',
-                           Constants.EVENT, 'notification for', event.name);
-      this.sendEventNotification(event);
-    });
-
-    this.on(Constants.CONNECTED, ({device, connected}) => {
-      DEBUG && console.log('AddonManagerProxy: Got',
-                           Constants.CONNECTED, 'notification for',
-                           device.title);
-      this.sendConnectedNotification(device, connected);
-    });
```

Comment:
This looks like its removing useful functionality that I didn't see added anyplace else.

Reply:
The event handler part was moved back into the gateway where it belonged: https://github.com/mozilla-iot/gateway/pull/2187/files#diff-8d2238f464e78e00c087a3f84b7a9786

The send part was moved into the device class: https://github.com/mozilla-iot/gateway-addon-node/pull/50/files#diff-2d36a01996aba0242ea152f771dd413c

Reply:
I see now.The emit in device.js was from the original gateway implementation when the plugins were all in-process.